### DISCUSSION TITLE
Help needed

### DISCUSSION THREAD

**Discussion by MohamedxSalah** [onitake/gsl-firmware#198] [Q&A]
well i gonna tell all my story to clarify that i'm kinda of noob

i installed an android x86 on my windows tablet , but the touch screen all missed up ( ex: when i touch the top left it registered the touch to bottom right , and some areas the touch isn't working on it )

so i google it :"D and i find out that the problem is with the "silead_ts.fw" , and that i must convert my windows "SileadTouch.fw" to "silead_ts.fw " 

and i remembered that when i did clean install of windows on my tablet i faced a problem where the axis was all missed up but hopefully i had a backup of my old windows drivers and sileadtouch.fw and as soon as i install it everything back to normal 

so long story short ,   i have some questions 
1. is my problem is really with the "silead_ts.fw" ? 
2. i noticed that in some cases the touch resolution maybe less than the display actual resolution , so my question is how to know my touch resolution on windows ? 

I attached my windows touch drivers along with  sileadtouch.fw  at the end of the post 

My tablet is Sary  8 inch model
display resolution is : 1200x800 
touch driver : ACPI\MSSL1680 

if someone could help me and do the port by himself i will be so thankful as i'm noob at linux and i'm rn downloading ubuntu specially for this thing 

[Touch.zip](https://github.com/onitake/gsl-firmware/files/1996715/Touch.zip)

**Comment by onitake** [comment]
Yes, the problem is almost certainly in the firmware or in the device parameters.
If any of them aren't correct, you will experience all kinds of problems with the touchscreen.

Unfortunately, fixing these problems is not easy, since it requires manual fine-tuning and a kernel driver modification to make all changes permanent. I just updated the readme with better instructions, but they may still be too difficult to understand if you don't have much Linux experience.

I'm afraid that I'm currently a bit too busy to provide support. Have you asked on xda-developers? Perhaps somebody might be able to help.

**Comment by MohamedxSalah** [comment]
sorry for late replay 
first i would thank you for your answer
2nd . i tried to make it alot of times but it's seems like that my problem is that i'm using my "screen resolution " for touch resolution and apparently they are different as the touch point still off " touch not working on all the screen and it's inverted " 

so my question is . how can i identify my touch screen resolution ?

**Comment by onitake** [comment]
As far as I know, there is no hardware register or firmware parameter that reports the resolution.
You'll have to experiment a bit to find it out. It's not terribly hard, just try some large values and narrow it down until you get it. It doesn't have to be precise, you can always finetune later.
